repos/os/src/drivers/spi/imx8q/ecspi/ecspi_driver.h

```diff
@@ -0,0 +1,122 @@
+/*
+ * \brief  Ecspi driver for imx8q
+ * \author Jean-Adrien Domage <jean-adrien.domage@gapfruit.com>
+ * \date   2021-04-20
+ */
+
+/*
+ * Copyright (C) 2013-2021 Genode Labs GmbH
+ * Copyright (C) 2021 gapfruit AG
+ *
+ * This file is part of the Genode OS framework, which is distributed
+ * under the terms of the GNU Affero General Public License version 3.
+ */
+
+#ifndef _INCLUDE__IMX8Q_ECSPI__DRIVER_H_
+#define _INCLUDE__IMX8Q_ECSPI__DRIVER_H_
+
+/* Genode includes */
+#include <base/env.h>
+#include <irq_session/client.h>
+#include <os/ring_buffer.h>
+#include <platform_session/connection.h>
+#include <timer_session/connection.h>
+#include <util/xml_node.h>
+
+/* Local includes */
+#include "spi_driver.h"
+#include "ecspi_mmio.h"
+
+namespace Spi {
+	using namespace Genode;
+
+	class Ecspi_driver;
+}
+
+
+class Spi::Ecspi_driver : public Spi::Driver {
+public:
+
+	struct Config {
+		bool     verbose       = false;
+		bool     loopback      = false;
+		uint8_t  clock_divider = 0;
+		uint64_t timeout       = 1000; /* in millisecond */
+	};
+
+
+private:
+
+	Genode::Env      &_env;
+	const Config      _config;
+	Timer::Connection _timer { _env };
```

Comment:
This timer is only used as a kind of wall clock currently. Please check with [[1]](https://genodians.org/m-stein/2021-05-07-short-guide-to-the-timeout-framework) and [[2]](https://genodians.org/m-stein/2021-05-31-a-short-guide-to-the-timer-session-interface) if the use of the timeout framework (and it's `curr_time()` function) fits your requirements.